src/platform_impl/linux/wayland/seat/pointer/mod.rs

```diff
@@ -42,6 +44,10 @@ impl PointerHandler for WinitState {
     ) {
         let seat = pointer.winit_data().seat();
         let seat_state = self.seats.get(&seat.id()).unwrap();
+        if seat_state.pointer.is_none() {
+            warn!("Received pointer event without pointer");
+            return;
+        }
```

Comment:
Instead of doing `is_none`, we should use a `match` here and out the result into a `let pointer` variable. That way we don't have to `unwrap` below.

Reply:
I refactored that way because I've assumed that it'll fail due to borrowing somehow, however it seems to work.